commons-datastore/commons-datastore-mongodb/src/main/java/org/opencb/commons/datastore/mongodb/MongoDBIterator.java

```diff
@@ -0,0 +1,39 @@
+package org.opencb.commons.datastore.mongodb;
+
+import com.mongodb.client.FindIterable;
+import com.mongodb.client.MongoCursor;
+
+import java.util.Iterator;
+
+public class MongoDBIterator<E> implements Iterator<E> {
+
+    private FindIterable<E> iterator;
+    private long numMatches;
+
+    public MongoDBIterator(FindIterable<E> iterator, long numMatches) {
+        this.iterator = iterator;
+        this.numMatches = numMatches;
+    }
+
+    @Override
+    public boolean hasNext() {
+        return iterator.iterator().hasNext();
+    }
+
+    @Override
+    public E next() {
+        return iterator.iterator().next();
+    }
+
+    public FindIterable<E> getIterable() {
+        return iterator;
+    }
+
+    public MongoCursor<E> iterator() {
+        return iterator.iterator();
+    }
```

Comment:
This method should not be necessary. We should call to .next() and .hasNext(). We would also need to implement close() method to close the iterator.

Reply:
Yes I agree! However, [MongoPersistentCursor.java](https://github.com/opencb/java-common-libs/blob/develop/commons-datastore/commons-datastore-mongodb/src/main/java/org/opencb/commons/datastore/mongodb/MongoPersistentCursor.java#L114) returns a `FindIterable<Document>`. Should I change to `MongoDBIterator`? I had a compile error with sort and batchsize so I'd need to add setters for those, I think?